modules/SofaMiscFem/src/SofaMiscFem/TriangularFEMForceField.inl

```diff
@@ -1173,9 +1173,10 @@ void TriangularFEMForceField<DataTypes>::computeStressAlongDirection(Real &stres
 template <class DataTypes>
 void TriangularFEMForceField<DataTypes>::computeStressAcrossDirection(Real &stress_across_dir, Index elementIndex, const Coord &dir, const type::Vec<3,Real> &stress)
 {
-    Index a = m_topology->getTriangle(elementIndex)[0];
-    Index b = m_topology->getTriangle(elementIndex)[1];
-    Index c = m_topology->getTriangle(elementIndex)[2];
+    const Triangle& tri = m_topology->getTriangle(elementIndex);
```

Comment:
I think m_topology->getTriangle() should be moved out of any loops (because the function is virtual). This can be done by changing the function signature to pass the triangles to the methods.